x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/percentile_ranks.tsx

```diff
@@ -0,0 +1,245 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiFormRow, EuiFieldNumberProps, EuiFieldNumber } from '@elastic/eui';
+import React, { useCallback } from 'react';
+import { i18n } from '@kbn/i18n';
+import { AggFunctionsMapping } from '@kbn/data-plugin/public';
+import { buildExpressionFunction } from '@kbn/expressions-plugin/public';
+import { OperationDefinition } from '.';
+import {
+  getFormatFromPreviousColumn,
+  getInvalidFieldMessage,
+  getSafeName,
+  isValidNumber,
+  getFilter,
+  isColumnOfType,
+  combineErrorMessages,
+} from './helpers';
+import { FieldBasedIndexPatternColumnMultipleValues } from './column_types';
+import { adjustTimeScaleLabelSuffix } from '../time_scale_utils';
+import { useDebouncedValue } from '../../../shared_components';
+import { getDisallowedPreviousShiftMessage } from '../../time_shift_utils';
+
+export interface PercentileRanksIndexPatternColumn
+  extends FieldBasedIndexPatternColumnMultipleValues {
+  operationType: 'percentile_ranks';
+}
+
+function ofName(name: string, value: number, timeShift: string | undefined) {
+  return adjustTimeScaleLabelSuffix(
+    i18n.translate('xpack.lens.indexPattern.percentileRanksOf', {
+      defaultMessage: 'Percentile rank ({value}) of {name}',
+      values: { name, value },
+    }),
+    undefined,
+    undefined,
+    undefined,
+    timeShift
+  );
+}
+
+const DEFAULT_PERCENTILE_RANKS_VALUE = 0;
+
+const supportedFieldTypes = ['number', 'histogram'];
+
+export const percentileRanksOperation: OperationDefinition<
+  PercentileRanksIndexPatternColumn,
+  'field',
+  { value: number }
+> = {
+  type: 'percentile_ranks',
+  displayName: i18n.translate('xpack.lens.indexPattern.percentileRanks', {
+    defaultMessage: 'Percentile ranks',
```

Comment:
Shouldn't it be "percentile rank" as it's just a single one?

Reply:
Done!